mongo/options/clientoptions.go

```diff
@@ -144,6 +145,20 @@ type IDPInfo struct {
 	RequestScopes []string
 }
 
+// CredentialsProvider is the function type that returns AWS credentials.
+type CredentialsProvider func(context.Context) (Credentials, error)
+
+// Credentials represents AWS credentials.
+type Credentials struct {
```

Comment:
Since this is specific to AWS credentials, this should be named `AWSCredentials`.